content/docs/reference/how.md

```diff
@@ -7,44 +7,89 @@ menu:
+
+The deployment engine is embedded into the `pulumi` CLI itself.
+
+## Resource Providers
+
+A resource provider is made up of two different pieces:
+
+1. A _resource plugin_, which is the binary used by the deployment engine to manage a resource. These plugins are stored in the _plugin cache_ (located in `~/.pulumi/plugins`) and can be manged with the `pulumi plugin` set of commands.
+2. An _SDK_ which provides bindings for each type of resource the provider can manage.
+
+Like the language runtime itself, the SDKs are available as regular packages.  For example, the for Node there is an `@pulumi/aws` package published to NPM and a `pulumi_aws` package for Python published to PyPI.  When these packages are added to your project, they run `pulumi plugin install` behind the scenes in order to download the resource plugin from pulumi.com.
+
+## Putting it all together
+
+Let's walk through a simple example, suppose we have the following Pulumi program, which creates two S3 buckets:
 
 ```javascript
-const bucket = new aws.s3.Bucket("media-bucket");
-const bucket = new aws.s3.Bucket("content-bucket");
+const mediaBucket = new aws.s3.Bucket("media-bucket");
+const contentBucket = new aws.s3.Bucket("content-bucket");
 ```
 
 Now, we run `pulumi stack init mystack`. Since `mystack` is a new stack, the "last deployed state" has no resources.
 
-Next, we run `pulumi up`. When the program runs to completion, it runs the two `new aws.s3.Bucket()` statements. So, the language host registers two resources with the engine.
+Next, we run `pulumi up`. Since this program is written in JavaScript, the Pulumi CLI launches the Node language host and requests that it execute the program. When the first `aws.s3.Bucket` object is constructed, the language host sends as _resource registration_ request to the deployment engine and then continues executing the program. This is subtle, but important: _When the call to `new aws.s3.Bucket` returns, it does not mean that the actual S3 bucket has been created in AWS_, it just means the language host has expressed that this bucket is part of the desired state of your infrastructure.  The language host continues to execute your program concurrently with the engine processing this request.
```

Comment:
Change "the language host sends as _resource registration_ " to "the language host sends a _resource registration_"